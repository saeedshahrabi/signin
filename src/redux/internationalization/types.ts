import { Action } from "redux";
import { TInternationalization } from "../../config/setup";


export enum LangType {
  CHANGE_APP_FLAG="CHANGE_APP_FLAG"
}

export interface IInternationalizationAction extends Action {
  payload: TInternationalization;
}