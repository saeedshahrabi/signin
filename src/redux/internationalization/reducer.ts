import { IInternationalizationAction, LangType } from "./types";
import { Setup, TInternationalization } from "../../config/setup";

export function LanguageReducer(state: TInternationalization,action: IInternationalizationAction): TInternationalization {
  switch (action.type) {
    case LangType.CHANGE_APP_FLAG:
      return action.payload;
  }
  if (state) {
    return state;
  }
  return Setup.internationalization;
}
