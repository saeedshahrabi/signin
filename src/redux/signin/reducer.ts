import { signin } from "./types";

const initialState = {
  loading: false,
  signin: [],
  error: "",
};

export const signinReducer = (state = initialState, action:any) => {
  switch (action.type) {
    case signin.SIGNIN_REQUEST:
      return {
        ...state,
        loading: true,
      };
    case signin.SIGNIN_SUCCESS:
      return {
        ...state,
        loading: false,
        signin: action.payload,
      };
    case signin.SIGNIN_FAILURE:
      return {
        ...state,
        loading: false,
        error: action.payload,
      };

    default:
      return state;
  }
};
