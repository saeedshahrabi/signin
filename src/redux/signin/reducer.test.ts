import { signinReducer } from "./reducer";
import { signin } from "./types";

describe("signin reduer", () => {
  it("when the request is send", () => {
    const action = { type: signin.SIGNIN_REQUEST };
    const expectedState = { loading: true, signin: [], error: "" };
    expect(signinReducer(undefined, action)).toEqual(expectedState);
  });
  it("when the request is success", () => {
    const action = { type: signin.SIGNIN_SUCCESS };
    const expectedState = {
      loading: false,
      signin: undefined,
      error: "",
    };
    expect(signinReducer(undefined, action)).toEqual(expectedState);
  });
  it("when the request is faild", () => {
    const action = { type: signin.SIGNIN_FAILURE };
    const expectedState = {
      loading: false,
      signin: [],
      error: undefined,
    };
    expect(signinReducer(undefined, action)).toEqual(expectedState);
  });
});
describe("snapshot testing for reducer", () => {
  test("get snap shot of signin reducer", () => {
    const action = { type: signin };
    expect(signinReducer(undefined, action)).toMatchSnapshot();
  });
});
