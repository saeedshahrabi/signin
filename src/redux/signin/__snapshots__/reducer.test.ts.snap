// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing for reducer get snap shot of signin reducer 1`] = `
Object {
  "error": "",
  "loading": false,
  "signin": Array [],
}
`;
